org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/commands/CheckBoundGrammarCommandTest.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.contentmodel.commands;
+
+import static org.eclipse.lemminx.XMLAssert.tde;
+import static org.eclipse.lemminx.XMLAssert.te;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.File;
+import java.util.concurrent.ExecutionException;
+
+import org.eclipse.lemminx.MockXMLLanguageServer;
+import org.eclipse.lemminx.extensions.contentmodel.commands.CheckBoundGrammarCommand;
+
+import org.eclipse.lemminx.utils.platform.Platform;
+import org.eclipse.lsp4j.TextDocumentEdit;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for associating a grammar to a given XML command.
+ *
+ * @author Angelo ZERR
+ *
+ */
+public class CheckBoundGrammarCommandTest {
+
+	@Test
+	public void checkDocumentWithoutBoundGrammar() throws InterruptedException, ExecutionException {
+		MockXMLLanguageServer languageServer = new MockXMLLanguageServer();
+
+		String xml = "<foo/>";
+		String xmlPath = getFileURI("src/test/resources/tag.xml");
+		TextDocumentIdentifier xmlIdentifier = languageServer.didOpen(xmlPath, xml);
+		String xsdPath = getFileURI("src/test/resources/xsd/tag.xsd");
+
+		Boolean actual = (Boolean) languageServer.executeCommand(CheckBoundGrammarCommand.COMMAND_ID, xmlIdentifier, xsdPath).get();
+		assertNotNull(actual);
+		assertEquals(true, actual);
+	}
+
+	@Test
+	public void checkDocumentWithBoundGrammar() throws InterruptedException, ExecutionException {
+		MockXMLLanguageServer languageServer = new MockXMLLanguageServer();
+
+		String xml = "<foo xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:noNamespaceSchemaLocation=\"tag.xsd\"/>";
+		String xmlPath = getFileURI("src/test/resources/tag.xml");
+		TextDocumentIdentifier xmlIdentifier = languageServer.didOpen(xmlPath, xml);
+		String xsdPath = getFileURI("src/test/resources/xsd/tag.xsd");
+
+		Boolean actual = (Boolean) languageServer.executeCommand(CheckBoundGrammarCommand.COMMAND_ID, xmlIdentifier, xsdPath).get();
+		assertNotNull(actual);
+		assertEquals(false, actual);
+	}
+
+	private static String getFileURI(String fileName) {
```

Comment:
I will write an another test with xml-model too, because in the future we should improve canBindGrammar to return true if the XML is only bound with xml-model, because you can add any xml-model. But we will see that in an another PR, for the moment please write a test with xml-model.